### PR TITLE
Add /tenancies endpoint and ListTenancies use case

### PR DESCRIPTION
**Pull request by zuren**
- Startup configures the dependency injection for the controller and the use case.
- The controller calls the ListTenancies use case.
- The ListTenancies use case uses it's injected ITenanciesGateway class to get tenancies by ref.

You can call the endpoint using:

```
curl http://localhost:32742/api/tenancies?tenancy_refs[0]=123&tenancy_refs[1]=456
```

### CODE REVIEW

File: LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs [hunk 1]
```diff
@@ -11,18 +13,198 @@ namespace LBHTenancyAPITest.Test.Controllers
...
+            {
+                Assert.Equal(expectedArguments, calledWith[0]);
+            }
+        }
+
+        private class ListTenanciesStub : IListTenancies
+        {
+            private readonly Dictionary<string, ListTenancies.ResponseTenancy> stubTenancies;
+
+            public ListTenanciesStub()
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs]: Can we separate these private classes out into their own file?

File: LBHTenancyAPI/UseCases/ListTenancies.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            var response = new Response();
+            var tenancies = tenanciesGateway.GetTenanciesByRefs(tenancyRefs);
+
+            response.Tenancies = tenancies.ConvertAll(tenancy => new ResponseTenancy()
+                {
+                    TenancyRef = tenancy.TenancyRef,
+                    LastActionCode = tenancy.LastActionCode,
+                    LastActionDate = string.Format("{0:u}", tenancy.LastActionDate),
+                    CurrentBalance = tenancy.CurrentBalance.ToString(),
```
Comment (StevenLeighton21) [inline LBHTenancyAPI/UseCases/ListTenancies.cs]: Could we use `ToString("C")` to format as a currency? Or is this the responsibility of whatever will display the value? "C" is localised currency using banker's rounding IIRC

File: LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs (hunk 1 above)
Comment (zuren) [reply LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs]: Good idea, they may be reusable elsewhere. My only concern would be them becoming more generic to fit more test cases, whereas if they stay here they'll stay very specific to the tests I have. Do you have a strong preference? I could go either way honestly.

**Comment by zuren** [comment]
Closed due to being merged into a separate PR, which has now merged into master.

**Pull request closed without merging**